ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2837,136 @@ public sealed class Bot : IAsyncDisposable {
 			}
 		}
 
+		private void OnInventoryChanged() {
+			Utilities.InBackground(CardsFarmer.OnNewItemsNotification);
+
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.DismissInventoryNotifications)) {
+				Utilities.InBackground(ArchiWebHandler.MarkInventory);
+			}
+
+			if (BotConfig.SendSetsOnCompleted) {
+				Utilities.InBackground(SendCompletedSets);
+			}
+		}
+
+		private async Task SendCompletedSets() {
+			IDocument? badgePage = await ArchiWebHandler.GetBadgePage(1).ConfigureAwait(false);
```

Comment:
This is prone to error of having more sets than one badge page can handle. You should get max pages similar to how `CardsFarmer` is doing that and ask for each page until you hit one that doesn't have any more sets - wishlist idea, detect if current page has less sets than maximum, and avoid additional request.

Reply:
This actually has TODO comment from me that I overlooked. Consider it done in next commit